backend/api_handler.py

```diff
@@ -146,10 +155,50 @@ def handler(event, context):
             'schema': SCHEMA,
         }
 
+        # Determine if there are any Operations that Require ReAuth From SSM Parameter
+        try:
+            reauth_apis = ParameterStoreManager.get_parameter_value(region=os.getenv('AWS_REGION', 'eu-west-1'), parameter_path=f"/dataall/{ENVNAME}/reauth/apis").split(',')
+            print("SSM", reauth_apis)
+        except Exception as e:
+            reauth_apis = None
     else:
         raise Exception(f'Could not initialize user context from event {event}')
 
     query = json.loads(event.get('body'))
+
+    # If The Operation is a ReAuth Operation - Ensure A Non-Expired Session or Return Error
+    if reauth_apis and query.get('operationName', None) in reauth_apis:
+        now = datetime.datetime.now(datetime.timezone.utc)
+        try:
+            auth_time_datetime = datetime.datetime.fromtimestamp(int(claims["auth_time"]), tz=datetime.timezone.utc)
+            if auth_time_datetime + datetime.timedelta(minutes=TTL) < now:
+                raise Exception("ReAuth")
+        except Exception as e:
+            print(f'ReAuth Required, Error: {e}')
+            response = {
+                "data": {query.get('operationName', 'operation') : None},
+                "errors": [
+                    {
+                        "message": f"ReAuth Required To Perform This Action {query.get('operationName', '')}",
+                        "locations": None,
+                        "path": [query.get('operationName', '')],
+                        "extensions": {
+                            "code": "REAUTH"
+                        }
+                    }
```

Comment:
If this is going to be logged, as a data platform admin I would like to know which user is needing the re-auth for audit reasons